File: Day1/invertBinaryTree_Donghee.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    
    def invertHelper(self, root):
        if (root == NULL):
            return root
        
        temp = root.left
        root.left = invertHelper(root.right)
        root.right = invertHelper(temp)
        
        
        return root
        
    
    def invertTree(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        return invertHelper(root)
        

File: Day1/invertBinaryTree_Donghee.cpp
// 6/1/2020
// Donghee Lee

class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        return invertHelper(root);
    }
    
    TreeNode* invertHelper(TreeNode* root) {
        if (root == nullptr) return nullptr;
        
        TreeNode* temp = root->left;
        root->left = invertHelper(root->right);
        root->right = invertHelper(temp);
        
        return root;
    }
};